packages/data/lib/src/request/api_request_representable.dart

```diff
@@ -0,0 +1,29 @@
+abstract class HTTPMethod<M> {}
+
+abstract class DioMethods implements HTTPMethod<String> {
+  static const String get = 'GET';
+
+  static const String post = 'POST';
+
+  static const String delete = 'DELETE';
+
+  static const String patch = 'PATCH';
+
+  static const String put = 'PUT';
```

Comment:
in case u want to keep this constants, it's better to refer to dio consts, like

`static const String get = dio.methodsConsts.get;`

why? tomorrow dio will publish the update with methods renaming and your class will become useless

Reply:
hmm, in Dio library I can't to find this constants... in whole library I faced smth like this:

> this.method = method ?? 'GET';

in one word it hardcoded anywhere:

> @override
>   Future<Response<T>> post<T>(
>     String path, {
>     data,
>     Map<String, dynamic>? queryParameters,
>     Options? options,
>     CancelToken? cancelToken,
>     ProgressCallback? onSendProgress,
>     ProgressCallback? onReceiveProgress,
>   }) {
>     return request<T>(
>       path,
>       data: data,
>       options: checkOptions('POST', options),
seems like dio developers not going to change this approach =) And I have no idea how can achieve this